lib/rubocop/cop/rspec/factory_bot/static_attribute_defined_dynamically.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      module FactoryBot
+        # Prefer declaring static attribute values without a block.
+        #
+        # @example
+        #   # bad
+        #   kind { :static }
+        #
+        #   # good
+        #   kind :static
+        #
+        #   # bad
+        #   comments_count { 0 }
+        #
+        #   # good
+        #   comments_count 0
+        #
+        #   # bad
+        #   type { User::MAGIC }
+        #
+        #   # good
+        #   type User::MAGIC
+        class StaticAttributeDefinedDynamically < Cop
+          MSG = 'Dont use a block to set a static value to an attribute.'.freeze
+
+          def_node_matcher :block_ast_method, <<-PATTERN
+          (block (send nil? _) _ ($_ ...))
+          PATTERN
+
+          def_node_search :factory_attributes, <<-PATTERN
+          (block (send nil? { :factory :trait } ...) _ { (begin $...) $(send ...) $(block ...) } )
+          PATTERN
+
+          def on_block(node)
+            return if node.method_name == :trait
+            factory_attributes(node).to_a.flatten.each do |attribute|
+              next if [nil, :send].include? block_ast_method(attribute)
```

Comment:
If you follow the approach above, you can start yielding the attributes:

```ruby
block_ast_method(attribute) do |attribute_value|
   add_offense(attribute_value, location: :expression)
end
```

Reply:
I changed it to just a `next` operator instead. Reads a little better IMO:

69ca2dac1e1c65fa6dc2290c7de52167951e7815